### ISSUE TITLE
When using scm or delivering, show link in report page

### ISSUE THREAD

**Issue by thibaultduponchelle** [melezhik/sparky#39]
When building from scm, or delivering things during build, put links in report page.

**Comment by melezhik** [comment]
so, you want an SCM repo link on a build report, page, right?

**Comment by melezhik** [comment]
what is delivering?

**Comment by thibaultduponchelle** [comment]
Yes.

Delivering : push tarballs somewhere for production deployment later

**Comment by melezhik** [comment]
So you suggest if a build produces any sort of tarball (for the future delivery/deployment) - add link to it as well? Those ones are not defined though a "spec" ( in comparison with scm which is a part of sparky.yaml ), the maximum we could do is to convert text into html links in a report. Or I miss your point?

**Comment by thibaultduponchelle** [comment]
I will answer later better, it is between plug-in and UI. You can upload artefacts to github for instance : https://dev.to/davorg/archiving-cpan-logs-in-github-actions-2hfd or you can upload to artifactory, ftp, s3, whatever and push link to the CI system.

**Comment by thibaultduponchelle** [comment]
I think the key is more about how plugins can pass extra infos to UI. I guess that currently it only returns logs.

**Comment by melezhik** [comment]
that's right. it's just a text file which is an output from a build process, no other structured info, that could be processed later in UI ...

**Comment by thibaultduponchelle** [comment]
Can we think about returning structured info ? Like a json containing a key "log:" + eventually some others ?

It would maybe requires changes almost everywhere ?

**Comment by melezhik** [comment]
underlying spark job is just [sparrowdo](https://github.com/melezhik/sparrowdo) runs against either localhost/docker or ssh host that writes to file. For example:

```
sparrowdo --localhost > report.log
```

sparrowdo is not designed to return structured info, so it'd be hard to do ... 

I don't know. Maybe I'll come up with _some_ solutions later ...
